### PR TITLE
Provide builtin functions even in loose assembly mode.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by leonardoalt** [comment]
Test failing

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 1]
```diff
@@ -50,14 +51,19 @@ struct CopyTranslate: public yul::ASTCopier
...
+		m_dialect(_dialect), m_context(_context), m_references(_references) {}
 
 	using ASTCopier::operator();
 
 	yul::YulString translateIdentifier(yul::YulString _name) override
 	{
-		return yul::YulString{"usr$" + _name.str()};
+		// Strictly, we should check that this builtin also exists in the target dialect,
+		// but currently, eveything is EVM anyway.
+		if (m_dialect.builtin(_name))
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:65]: Added this.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -50,14 +51,19 @@ struct CopyTranslate: public yul::ASTCopier
...
+	CopyTranslate(yul::Dialect const& _dialect, IRGenerationContext& _context, ExternalRefsMap const& _references):
+		m_dialect(_dialect), m_context(_context), m_references(_references) {}
 
 	using ASTCopier::operator();
 
 	yul::YulString translateIdentifier(yul::YulString _name) override
 	{
-		return yul::YulString{"usr$" + _name.str()};
+		// Strictly, we should check that this builtin also exists in the target dialect,
+		// but currently, eveything is EVM anyway.
```
Comment (leonardoalt) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp]: typo `eveything`

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:65]: Is this what the comment says it doesn't do currently?

Comment (chriseth) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:65]: What I was referring to was that the inline assembly snippet could use a different dialect than the dialect we are currently compiling to. So the current code basically assumes that all built-in functions in the assembly snippet also exist in the target Yul dialect and have the same name.

**Comment by chriseth** [comment]
Re-wrote the comment.

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/codegen/ir/IRGeneratorForStatements.cpp:65]: ah ok

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**